File: app/src/main/java/com/arcao/feedback/collector/LocusMapInfoCollector.java
package com.arcao.feedback.collector;

import android.content.Context;
import locus.api.android.ActionTools;
import locus.api.android.utils.LocusInfo;
import locus.api.android.utils.LocusUtils;

public class LocusMapInfoCollector extends Collector {
	private final Context mContext;

	public LocusMapInfoCollector(Context context) {
		this.mContext = context;
	}

	@Override
	public String getName() {
		return "LocusMapInfo";
	}

	@Override
	protected String collect() {
		StringBuilder sb = new StringBuilder();

		try {
			LocusUtils.LocusVersion lv = LocusUtils.getActiveVersion(mContext);
			if (lv != null) {
				sb.append("Locus Version = ").append(lv.getVersionName());
				sb.append("\nLocus Package = ").append(lv.getPackageName());

				LocusInfo info = ActionTools.getLocusInfo(mContext, lv);
				if (info != null) {
					sb.append("\nLocus info:\n").append(info.toString());
				}
			} else {
				sb.append("Locus not installed!");
			}
		} catch (Throwable t) {
			sb.append("Unable to get info from Locus Map:\n").append(throwableToString(t));
		}

		return sb.toString();
	}
}

File: app/src/main/java/com/arcao/feedback/collector/LocusMapInfoCollector.kt
package com.arcao.feedback.collector

import android.content.Context
import locus.api.android.ActionTools
import locus.api.android.utils.LocusUtils

class LocusMapInfoCollector(private val context: Context) : Collector() {
    override val name: String
        get() = "LocusMapInfo"

    override suspend fun collect(): String {
        val sb = StringBuilder()

        try {
            val lv = LocusUtils.getActiveVersion(context)
            if (lv != null) {
                sb.append("Locus Version = ").append(lv.versionName)
                sb.append("\nLocus Package = ").append(lv.packageName)

                val info = ActionTools.getLocusInfo(context, lv)
                if (info != null) {
                    sb.append("\nLocus info:\n").append(info.toString())
                }
            } else {
                sb.append("Locus not installed!")
            }
        } catch (e: Exception) {
            sb.append("Unable to get info from Locus Map:\n").append(throwableToString(e))
        }

        return sb.toString()
    }
}
